### ISSUE TITLE
feat(external-link): respect Retry-After header on HTTP 429 responses

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#269]
## Overview

When an external link check receives an HTTP 429 (Too Many Requests), gomarklint currently treats it as a passing status and returns immediately without retrying. This means rate-limited URLs are silently skipped rather than re-checked after the server's requested delay.

Adding `Retry-After` support allows gomarklint to wait the server-specified duration and retry, producing accurate results instead of a silent pass.

## Problem

GitHub and many other major hosts enforce rate limits and return 429 with a `Retry-After` header. A documentation repo with many GitHub links will frequently hit this limit in CI, causing those links to be silently skipped — potentially hiding real broken links.

Current behavior (`internal/rule/external_link.go`):

```go
// 429 is in defaultAllowedStatuses → returns immediately, no retry
var defaultAllowedStatuses = []int{http.StatusTooManyRequests}

if err == nil && isAllowedStatus(status, allowedStatuses) {
    return status, nil  // silent pass, no retry
}
```

### Files to modify

- `internal/rule/external_link.go`
- `internal/config/config.go` (new option: `retry_after_max_seconds`)

### Changes

**1. `performCheck` — return the `Retry-After` header value**

Change signature from `(int, error)` to `(int, string, error)` so the header value reaches `checkURL`.

```go
func performCheck(client *http.Client, url string) (int, string, error)
```

**2. Add `parseRetryAfter` helper**

`Retry-After` is either a delay in seconds or an HTTP-date. Parse both forms into a `time.Duration`, capped at a configurable maximum (default: 60 s) to prevent CI hangs.

```go
func parseRetryAfter(header string, maxSeconds int) time.Duration
```

**3. `checkURL` — handle 429 with Retry-After**

```go
status, retryAfter, err = performCheck(client, url)

if status == 429 {
    wait := parseRetryAfter(retryAfter, maxRetryAfterSeconds)
    time.Sleep(wait)
    continue  // retry
}
```

If still 429 after all retries, treat as a passing status (rate limiting ≠ broken link) — same as current behavior.

**4. New config option**

```json
{
  "rules": {
    "external-link": {
      "enabled": true,
      "retry_after_max_seconds": 60
    }
  }
}
```

## Retry-After header format

The header value is one of two forms per RFC 9110:

| Form | Example |
|---|---|
| Delay in seconds | `Retry-After: 30` |
| HTTP-date | `Retry-After: Thu, 08 May 2026 06:00:00 GMT` |

Both must be handled. If the header is absent or unparseable, fall back to the existing fixed retry delay (`retry_delay_ms`).

## Acceptance criteria

- [ ] 429 with `Retry-After: <seconds>` → waits that many seconds (capped), then retries
- [ ] 429 with `Retry-After: <http-date>` → waits until that time (capped), then retries  
- [ ] 429 with no `Retry-After` header → falls back to existing `retry_delay_ms` backoff
- [ ] 429 after all retries exhausted → passes (not reported as broken link)
- [ ] `retry_after_max_seconds` config option respected
- [ ] Unit tests for `parseRetryAfter` covering both header forms, missing header, and cap enforcement
